main.rb

```diff
@@ -0,0 +1,160 @@
+# enumerable methods
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i]
+      i += 1
+    end
+    my_arr
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i], i
+      i += 1
+    end
+    my_arr
+  end

```

Comment:
- [ ] my_each_with_index when a block is given when self is a range after calling the given block once for each element in self, returns the range itself. Your method is returning an array, even when calling my_each_with_index from a range

Example:
`(1..3).my_each_with_index { | value, index | puts index }` should print 0, 1 and 2, and after that should return the range itself `1..3`

Print from the original method:

![Screenshot from 2020-06-05 09-22-33](https://user-images.githubusercontent.com/30977845/83875778-1a3ff000-a70e-11ea-9d1a-d2d00353a787.png)